### PR TITLE
Update to PHPStan 0.10

### PR DESCRIPTION
**Pull request by carusogabriel**
The 3 remain errors sound like false-positives :thinking:

### CODE REVIEW

**Comment by Majkl578** [comment]
Needs to use phpstan.neon.dist.
Needs to use level + path in config.
.travis.yml needs to be updated to reflect the changes.

**Comment by Majkl578** [comment]
> The 3 remain errors sound like false-positives

https://github.com/phpstan/phpstan/issues/1098

Feel free to put them into _ignored errors_.

**Comment by Majkl578** [comment]
Please also add `phpstan.neon` into .gitignore. Then it's good to go. :+1:

**Comment by carusogabriel** [comment]
> Please also add `phpstan.neon` into `.gitignore`. Then it's good to go. +1

@Majkl578 Do we do it in other repos? Is there any reason why?

**Comment by Majkl578** [comment]
Yes, same reason as phpunit.xml or phpcs.xml.

**Comment by carusogabriel** [comment]
@Majkl578 Should be RTM :)

File: lib/Doctrine/Inflector/Rules/Irregular.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function getRules() : iterable
 
     public function inflect(string $word) : string
     {
-        if (preg_match('/(.*)\\b(' . $this->getRegex() . ')$/i', $word, $regs)) {
+        if (preg_match('/(.*)\\b(' . $this->getRegex() . ')$/i', $word, $regs) > 0) {
```
Comment (Majkl578) [inline lib/Doctrine/Inflector/Rules/Irregular.php]: This should be `=== 1`, but it's just a nitpick with no functional difference.

File: phpstan.neon.dist [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    - vendor/phpstan/phpstan-strict-rules/rules.neon
+
+parameters:
+    level: 7
+    paths:
+      - lib
+      - tests
+
+    ignoreErrors:
+      - '#Parameter \#1 ...\$(rules|words) of class Doctrine\\Inflector\\Rules\\(Uninflected|Irregular) constructor expects array<int, Doctrine\\Inflector\\Rules\\(Rule|Word)>, iterable given.#'
```
Comment (Majkl578) [inline phpstan.neon.dist]: Hint: If you pick different delimiter than `#`, you wouldn't need to escape it in the string. My favourite is `~`.

**Pull request merged**